aspnetcore/fundamentals/http-requests/samples/6.x/HttpRequestsSample/Pages/Consumption/NamedAndTypedClient.cshtml.cs

```diff
@@ -0,0 +1,38 @@
+using HttpRequestsSample.GitHub;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Extensions.Http;
+
+namespace HttpRequestsSample.Pages;
+
+// <snippet_Class>
+public class NamedAndTypedClientModel : PageModel
+{
+    private readonly IHttpClientFactory _httpClientFactory;
+    private readonly ITypedHttpClientFactory<GitHubService> _gitHubServiceFactory;
+
+    public NamedAndTypedClientModel(IHttpClientFactory httpClientFactory, 
+        ITypedHttpClientFactory<GitHubService> gitHubServiceFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+        _gitHubServiceFactory = gitHubServiceFactory;
+    }
+
+    public IEnumerable<GitHubBranch>? GitHubBranches { get; set; }
+
+    public async Task OnGet()
+    {
+        try
+        {
+            var quickClient = _httpClientFactory.CreateClient("Quick");
+            var gitHubService = _gitHubServiceFactory.CreateClient(quickClient);
+            GitHubBranches = await gitHubService.GetAspNetCoreDocsBranchesAsync();
+        }
+        catch (TaskCanceledException ex) when (ex.InnerException is TimeoutException)
+        {
+            var tolerantClient = _httpClientFactory.CreateClient("LatencyTolerant");
```

Comment:
This exact scenario should be handled not by multiple clients, but by resilience strategies applied via Polly